tests/TableViewTest.cc

```diff
@@ -0,0 +1,224 @@
+    ProducerConfiguration producerConfiguration;
+    producerConfiguration.setSchema(schemaInfo);
+    Producer producer;
+    ASSERT_EQ(ResultOk, client.createProducer(topic, producerConfiguration, producer));
+
+    // Create table view failed, The schema is not compatible
+    TableViewConfiguration tableViewConfiguration{.schemaInfo = SchemaInfo(AVRO, "", "")};
+    TableView tableView;
+    ASSERT_EQ(ResultIncompatibleSchema, client.createTableView(topic, tableViewConfiguration, tableView));
+    ASSERT_EQ(ResultConsumerNotInitialized, tableView.close());
+
+    // Create table view success.
+    ASSERT_EQ(ResultOk, client.createTableView(topic, {.schemaInfo = schemaInfo}, tableView));
+    ASSERT_EQ(ResultOk, tableView.close());
+
+    // Test async create and close the client during the process.
+    Latch latch(1);
+    client.createTableViewAsync(
+        topic, tableViewConfiguration, [&latch](Result result, const TableView& tableView) {
+            latch.countdown();
+            ASSERT_TRUE(result == ResultDisconnected || result == ResultAlreadyClosed);
+        });
+    client.close();
+    latch.wait();
+}
+
+TEST(TableViewTest, testSimpleTableView) {
+    const std::string topic = "testTableView" + std::to_string(time(nullptr));
+    Client client(lookupUrl);
+
+    ProducerConfiguration producerConfiguration;
+    Producer producer;
+    ASSERT_EQ(ResultOk, client.createProducer(topic, producerConfiguration, producer));
+
+    auto count = 20;
+    for (int i = 0; i < count; ++i) {
+        auto msg = MessageBuilder()
+                       .setPartitionKey("key" + std::to_string(i))
+                       .setContent("value" + std::to_string(i))
+                       .build();
+        ASSERT_EQ(ResultOk, producer.send(msg));
+    }
+
+    // Create table view and assert size.
+    TableView tableView;
+    ASSERT_EQ(ResultOk, client.createTableView(topic, {}, tableView));
+    ASSERT_EQ(tableView.size(), count);
+
+    // Send some more messages, The 0 ~ count message key/value is duplicated send.
+    for (int i = 0; i < count * 2; ++i) {
+        auto msg = MessageBuilder()
+                       .setPartitionKey("key" + std::to_string(i))
+                       .setContent("value" + std::to_string(i))
+                       .build();
+        ASSERT_EQ(ResultOk, producer.send(msg));
+    }
+    waitUntil(
+        std::chrono::seconds(2), [&] { return tableView.size() == count * 2; }, 1000);
+    ASSERT_EQ(tableView.size(), count * 2);
+
+    // assert interfaces.
+    std::string value;
+    ASSERT_TRUE(tableView.getValue("key1", value));
+    ASSERT_EQ(value, "value1");
+    ASSERT_TRUE(tableView.retrieveValue("key1", value));
+    ASSERT_EQ(value, "value1");
+    ASSERT_FALSE(tableView.containsKey("key1"));
+    ASSERT_EQ(tableView.snapshot().size(), count * 2 - 1);
+    ASSERT_EQ(tableView.size(), 0);
+
+    client.close();
+}
+
+TEST(TableViewTest, testPublishNullValue) {
```

Comment:
```suggestion
TEST(TableViewTest, testPublishEmptyValue) {
```

Actually the C++ client does not support sending a "null" value.